File: CS/Scaffolding.ManyToMany/Data/SchoolContext.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Data.Entity;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Scaffolding.ManyToMany.Model {
    public class Student {
        public int StudentId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }

        public virtual ICollection<Course> CoursesAttending { get; set; }

        public Student() {
            CoursesAttending = new HashSet<Course>();
        }
    }

    public class Course {
        public int CourseId { get; set; }
        public string Title { get; set; }

        public virtual ICollection<Student> Students { get; set; }

        public Course() {
            Students = new HashSet<Student>();
        }
    }

    public class SchoolContext : DbContext {
        public DbSet<Course> Courses { get; set; }
        public DbSet<Student> Students { get; set; }
        public SchoolContext() {
        }
    }
}

File: VB/Scaffolding.ManyToMany/Data/SchoolContext.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema
Imports System.Data.Entity
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace Scaffolding.ManyToMany.Model
	Public Class Student
		Public Property StudentId() As Integer
		Public Property FirstName() As String
		Public Property LastName() As String

		Public Overridable Property CoursesAttending() As ICollection(Of Course)

		Public Sub New()
			CoursesAttending = New HashSet(Of Course)()
		End Sub
	End Class

	Public Class Course
		Public Property CourseId() As Integer
		Public Property Title() As String

		Public Overridable Property Students() As ICollection(Of Student)

		Public Sub New()
			Students = New HashSet(Of Student)()
		End Sub
	End Class

	Public Class SchoolContext
		Inherits DbContext

		Public Property Courses() As DbSet(Of Course)
		Public Property Students() As DbSet(Of Student)
		Public Sub New()
		End Sub
	End Class
End Namespace
